sdk/identity/azure-identity-v2/src/main/java/com.azure.identity.v2/CredentialBuilderBase.java

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.v2;
+
+import com.azure.identity.v2.implementation.IdentityClientOptions;
+import com.azure.identity.v2.implementation.util.ValidationUtil;
+import io.clientcore.core.instrumentation.logging.ClientLogger;
+import io.clientcore.core.utils.configuration.Configuration;
+
+
+/**
+ * The base class for all the credential builders.
+ * @param <T> the type of the credential builder
+ */
+public abstract class CredentialBuilderBase<T extends CredentialBuilderBase<T>> {
+    private static final ClientLogger LOGGER = new ClientLogger(CredentialBuilderBase.class);
+
+    String clientId;
+    String tenantId;
```

Comment:
what if we lift these into `IdentityClientOptions`? We pass that into all the client types anyway. Maybe loose fields in the builders that just end up getting passed next to the bag should always live there.